numpy/core/src/multiarray/ctors.c

```diff
@@ -3629,6 +3648,16 @@ array_from_text(PyArray_Descr *dtype, npy_intp num, char *sep, size_t *nread,
         }
     }
     NPY_END_ALLOW_THREADS;
+
+    if (end_not_reached) {
+        /* 2019-05-21, NumPy 1.18  */
+        if (DEPRECATE(
+            "string or file could not be read to its end due to unmatched "
+            "data; this raise an error in the future.") < 0) {
```

Comment:
this **will** raise

Reply:
Changed (and one style thing). I wasn't sure though if we should be putting in versions. And I guess I put 1.18 for now above, but can change (and when we decided write release notes quickly).